### PR TITLE
Unity catalog exporter: tests

### LINKED ISSUE #6966: Unity catalog table registration hook
Using the output of the Delta Lake catalog exporter (#6965), configure an external table in Databricks' Unity catalog, where the schema is the branch name.

A minimal design can be found [here](https://github.com/treeverse/lakeFS/tree/master/design/open/unity-catalog-exporter.md).

A part of https://github.com/treeverse/lakeFS/issues/6771

### PR DESCRIPTION
**Pull request by Jonathan-Rosenberg**
Closes #6966

### CODE REVIEW

File: pkg/actions/testdata/lua/catalogexport_unity.lua [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            ["my_table3_failure"] = {
+                status = "FAILED",
+            },
+        },
+    },
+}
+
+-- Loads a mock table (descriptor) extractor
+local function load_table_descriptor(tables)
+    package.loaded["lakefs/catalogexport/table_extractor"] = {
```
Comment (Isan-Rivkin) [inline pkg/actions/testdata/lua/catalogexport_unity.lua:92]: love that package.loaded solution, I'll change that in glue exporter at some point for the lakeFS client!

File: pkg/actions/lua/lakefs/catalogexport/unity_exporter.lua [hunk 2]
```diff
@@ -42,6 +42,9 @@ local function register_tables(action, table_descriptors_path, delta_table_paths
             error("catalog name is required to proceed with unity catalog export")
         end
         local schema_name = databricks_client.create_or_get_schema(branch_id, catalog)
+        if not schema_name then
+            error("failed creating/getting catalog's schema")
```
Comment (Isan-Rivkin) [inline pkg/actions/lua/lakefs/catalogexport/unity_exporter.lua:46]:
```suggestion
            error("failed creating/getting catalog's schema: ", catalog, branch_id)
```

**Review by Isan-Rivkin — CHANGES_REQUESTED** [review]
Nice, minor comment!

File: pkg/actions/lua/lakefs/catalogexport/unity_exporter.lua (hunk 2 above)
Comment (Jonathan-Rosenberg) [reply pkg/actions/lua/lakefs/catalogexport/unity_exporter.lua]: done

**Review by Isan-Rivkin — APPROVED** [review]
LGTM

**Pull request merged**